restapi/src/main/java/io/stargate/web/docsapi/service/DocumentService.java

```diff
@@ -1217,11 +1213,6 @@ private JsonNode searchWithOnlyInMemoryFilters(
               .collect(Collectors.toList());
     }
 
-    if (!inMemoryFilters.isEmpty() && paginator.hasDbPageState()) {
-      throw new DocumentAPIRequestException(
-          "The results as requested must fit in one page, try increasing the `page-size` parameter.");
```

Comment:
Is this no longer an issue? It is taking me some extra efforts to implement on my branch :smiley: So, I'll wait for this to merge and rebase my refactoring.

Reply:
So - I realized along the way that this can be left out - the page size for searching within a document just works in a strange way for limited support filters as it is...but I think it is fine to remove this behavior, it is too restrictive as it is